src/bridge/IrcHandler.ts

```diff
@@ -623,16 +623,29 @@ export class IrcHandler {
             this.registeredNicks[nickKey] = true;
         }
 
-        const promises = [];
+        const failed = [];
         for (const room of matrixRooms) {
             req.log.info(
                 "Relaying in room %s", room.getId()
             );
-            promises.push(
-                this.ircBridge.sendMatrixAction(room, virtualMatrixUser, mxAction)
-            );
+            try {
+                await this.ircBridge.sendMatrixAction(room, virtualMatrixUser, mxAction);
+            }
+            catch (ex) {
+                // Check if it was a permission fail.
+                if (ex.data?.errcode === "M_FORBIDDEN" &&
+                    ex.data?.error?.startsWith("You don't have permission to post that to the room.")) {
```

Comment:
Is this string required in homeservers from the spec?
Comparing error messages seems like it's testing against a specific implementation. Is the error code not enough?

Reply:
The problem is they might also not be able to join the room because the join rules have changed, which we don't want to treat the same way. It's hard though, because there isn't an easy way in the spec to distinguish.